backend/de.metas.material/dispo-service/src/main/java/de/metas/material/dispo/service/candidatechange/handler/DemandCandiateHandler.java

```diff
@@ -147,9 +148,31 @@ public CandidateSaveResult onCandidateNewOrChange(
 			fireSupplyRequiredEventIfNeeded(candidateSaveResult.getCandidate(), savedStockCandidate);
 		}
 
+		if (candidateSaveResult.getQtyDelta().signum() < 0)
+		{
+			fireSupplyRequiredDecreasedEventIfNeeded(savedCandidate, candidateSaveResult.getQtyDelta().negate());
+		}
+
 		return candidateSaveResult;
 	}
 
+	private void fireSupplyRequiredDecreasedEventIfNeeded(final Candidate savedCandidate, final BigDecimal decreasedQty)
+	{
+		if (isCandidateEligibleForDecrease(savedCandidate))
```

Comment:
what do u mean  by "is eligible"?
what's the case when it's not eligible?

Reply:
Initially I was imagining some more logic to prevent creating unnecessary events from being created. But with the lazy retrieval of the PPOrderCandidateIds/DDOrderCandidateIds/PurchaseCandidateIds, it does not make sense anymore. removed